src/infrastructure/NetworkHttp.ts

```diff
@@ -58,4 +63,18 @@ export class NetworkHttp extends Http implements NetworkRepository {
             catchError((error) =>  throwError(this.errorHandling(error)))),
         );
     }
+
+    /**
+     * Get current network type name and description
+     *
+     * @return current network type name and description
+     */
+    public getNetworkName(): Observable<NetworkName> {
+        return observableFrom(this.networkRouteApi.getNetworkType()).pipe(
+            map((response: { response: ClientResponse; body: NetworkTypeDTO; }) => {
+                return new NetworkName(response.body.name, response.body.description);
+            }),
+            catchError((error) =>  throwError(this.errorHandling(error))),
```

Comment:
Remove the inner response. Use the `body.name` and `body.description` directly. It's type-safe. You don't need the `: NetworkTypeDto`, ts automatically knows that the body is typed to that NetworkTypeDto. 

You can also remove curly braces and the return world

I have the same method in my PR. I called NetworkInfo with a name and description. 

https://github.com/nemtech/nem2-sdk-typescript-javascript/pull/342/commits/3637bff3f7a66db7d3c169a34c417b1fb4920880#diff-5cb1ce66c671496f743ab2481416cf08R73

Reply:
will apply on rest HTTP clients in other PR